spark/spark-tensorflow-connector/src/main/scala/org/tensorflow/spark/datasources/tfrecords/serde/DefaultTfRecordRowEncoder.scala

```diff
@@ -111,9 +124,20 @@ object DefaultTfRecordRowEncoder extends TfRecordRowEncoder {
         FloatListFeatureEncoder.encode(ArrayData.toArrayData(row.get(index)).toFloatArray())
       case ArrayType(DoubleType, _) =>
         FloatListFeatureEncoder.encode(ArrayData.toArrayData(row.get(index)).toDoubleArray().map(_.toFloat))
+      case ArrayType(DecimalType(), _) =>
+        val decimalArray = ArrayData.toArrayData(row.get(index)).toArray[Decimal](DataTypes.createDecimalType())
+        FloatListFeatureEncoder.encode(decimalArray.map(_.toFloat))
       case ArrayType(_, _) =>
         BytesListFeatureEncoder.encode(ArrayData.toArrayData(row.get(index)).toArray[String](StringType))
-      case _ => BytesListFeatureEncoder.encode(Seq(row.getString(index)))
+      case VectorType => {
+        val field = row.get(index)
+        field match {
+          case v: SparseVector => FloatListFeatureEncoder.encode(v.toDense.toArray.map(_.toFloat))
```

Comment:
wouldn't `tf.VarLenFeature` be a better match for SparseVector? The `toDense` will be very inefficient for sparse features with large vocabulary.